File: 0013-roman-to-integer/0013-roman-to-integer.java
class Solution {
    public int romanToInt(String s) {
        if (s.length() == 0) {
            return 0;
        }
        int ans= getNum(s.charAt(0));
        for(int i=1;i<s.length();i++){
            if(getNum(s.charAt(i))>getNum(s.charAt(i-1))){
                ans+=getNum(s.charAt(i))- (2*getNum(s.charAt(i-1)));
            }else{
                ans+=getNum(s.charAt(i));
            }
           
        }
         return ans;
    }

    int getNum(char n) {

        switch (n) {

            case 'I':
                return 1;
            case 'V':
                return 5;
            case 'X':
                return 10;
            case 'L':
                return 50;
            case 'C':
                return 100;
            case 'D':
                return 500;
            case 'M':
                return 1000;
            default:
                return 0;

        }
    }
}

File: 0013-roman-to-integer/0013-roman-to-integer.py
class Solution:
    def romanToInt(self, s: str) -> int:
        roman = {
            'I': 1, 'V': 5, 'X': 10,
            'L': 50, 'C': 100,
            'D': 500, 'M': 1000
        }
        ans=0
        prev=0
        for ch in s:
            val=roman[ch]
            if val>prev:
                ans+=val-(2*prev)
            else:
                ans+=val
            prev=val
        return ans

